core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java

```diff
@@ -1482,9 +1490,23 @@ private void handleSqlBasicCallForTimestampMulti(SqlWriter writer, SqlCall call)
   private void unparseExtractFunction(SqlWriter writer, SqlCall call, int leftPrec, int rightPrec) {
     switch (call.operand(0).toString()) {
     case "EPOCH" :
-      final SqlWriter.Frame epochFrame = writer.startFunCall("UNIX_SECONDS");
-      call.operand(1).unparse(writer, leftPrec, rightPrec);
-      writer.endFunCall(epochFrame);
+      SqlNode firstOperand = call.operand(1);
+      if (firstOperand instanceof SqlBasicCall
+          && ((SqlBasicCall) firstOperand).getOperator().kind == SqlKind.MINUS) {
+        SqlNode leftOperand = ((SqlBasicCall) firstOperand).getOperands()[0];
+        SqlNode rightOperand = ((SqlBasicCall) firstOperand).getOperands()[1];
+        final SqlWriter.Frame epochFrame = writer.startFunCall("UNIX_SECONDS");
```

Comment:
Call to `UNIX_SECONDS` should be added inside `unparseOperandAsTimestamp()` only.

Reply:
No because we are unparsing leftOperand and rightOperand and they are wrapped inside unix_seconds